### PR TITLE
Update to SystemView v3.58

### PR DESCRIPTION
**Pull request by brycewilkins**
This repository has suffered a few setbacks over past years with Zephyr related patches being lost (causing things to break for RTT/SystemView/Zephyr), and it has been reported as hard to review PRs due to Windows style line endings. There is also no instruction on how to update from the existing Segger source code to a newer release, leading to partial updates that make subsequent updates more difficult and error prone.

This PR attempts to improve things by:
1. Changing all line endings to Unix style LF (`\n`)
2. Updating (completely) to SystemView v3.58 Target sources 
    - Note: #17 selectively introduced changes from SystemView v3.52a Target sources (many SYSVIEW changes skipped, files still referred to "SystemView version: 3.40", etc.)
4. Adding a README that identifies Segger "upstream" and suggests a procedure for updating to a new Segger release

To do:
- [x] Testing with SystemView application v3.58
Platform:
```
nRF Connect SDK v2.8.0-a2386bfc8401
Zephyr OS v3.7.99-0bc3393fb112

Board: Nordic Semi nRF52840 DK
```
image
- [x] Zephyr repo PR https://github.com/zephyrproject-rtos/zephyr/pull/82352

### CODE REVIEW

**Comment by brycewilkins** [comment]
@cfriedt The suggestion for [Unix style line endings](https://github.com/zephyrproject-rtos/segger/pull/17#issuecomment-2479999310) is accommodated in this PR.

@henrikbrixandersen @nordic-krch @nordicjm Hope you are able to have a look.

Thank you!

**Review by nordic-krch — APPROVED** [review]

**Comment by brycewilkins** [comment]
@cfriedt @henrikbrixandersen Would you be able to have a look? Thank you!

File: README.md
Comment (henrikbrixandersen) [inline README.md:1]: Please break the long lines (e.g. at 100 chars) to improve readability.

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]
Looks good, one small nit.

File: README.md
Comment (brycewilkins) [reply README.md:1]: Hi @henrikbrixandersen , thanks for the review!

The long URLs create some challenges. What do you recommend as the approach for this:

```
This repository is the Zephyr project module for integrating Segger [RTT](https://www.segger.com/products/debug-probes/j-link/technology/about-real-time-transfer) ([wiki](https://wiki.segger.com/RTT)), [SystemView](https://www.segger.com/products/development-tools/systemview/) ([wiki](https://wiki.segger.com/SystemView)) and [Monitor mode debugging](https://www.segger.com/products/debug-probes/j-link/technology/monitor-mode-debugging/) with Zephyr RTOS.
```

One option:
```
This repository is the Zephyr project module for integrating Segger
[RTT](https://www.segger.com/products/debug-probes/j-link/technology/about-real-time-transfer)
([wiki](https://wiki.segger.com/RTT)),
[SystemView](https://www.segger.com/products/development-tools/systemview/)
([wiki](https://wiki.segger.com/SystemView)) and
[Monitor mode debugging](https://www.segger.com/products/debug-probes/j-link/technology/monitor-mode-debugging/)
with Zephyr RTOS.
```
but that still exceeds 100 chars on one line.

I could move all the URLs to the end of the document in a **References** section.

Comment (henrikbrixandersen) [reply README.md:1]: Yes, you could convert them to Markdown reference-style links. That would improve readability of the raw file too, I think. 

@kartben Any opinion here?

Comment (brycewilkins) [reply README.md:1]: @henrikbrixandersen Neater now! 4a7b86ebb861b83adb12ae2cd791f69115cd49b8

And all lines under 100 chars to boot!

**Review by henrikbrixandersen — APPROVED** [review]

**Review by kartben — APPROVED** [review]

**Pull request merged**